### PR TITLE
guide: split DVC Files/Internals (prep for dvc.yaml 2.0 updates)

### PR DESCRIPTION
**Pull request by jorgeorpinel**
Also includes

- Separate **Outputs** and **Dependencies** sections in **DVC Files** guide

Next up:

- Review "metafile" terminology.
- Introduce basics from https://github.com/iterative/dvc/wiki/Parametrization to the **DVC Files** guide
- Further split **DVC Files** into separate pages?

### CODE REVIEW

**Comment by jorgeorpinel** [comment]
@ivan most files just have link updates, please focus on content/docs/sidebar.json, content/docs/user-guide/dvc-files.md, and content/docs/user-guide/dvc-internals.md. The changes can be visualized in https://dvc-landing-dvc-yaml-pa-e4wbve.herokuapp.com/doc/user-guide

File: content/docs/user-guide/dvc-files.md [hunk 1]
```diff
@@ -20,8 +15,13 @@ Both `.dvc` files and `dvc.yaml` use human-friendly YAML 1.2 schemas, described
 below. We encourage you to get familiar with them so you may create, generate,
 and edit them on your own.
 
-Both the internal directory and these metafiles should be versioned with Git (in
-Git-enabled <abbr>repositories</abbr>).
+These metafiles should be versioned with Git (in Git-enabled
+<abbr>repositories</abbr>).
+
+See the [internals guide](/doc/user-guide/dvc-internals) for the contents of the
+`.dvc/` directory.
```
Comment (jorgeorpinel) [inline content/docs/user-guide/dvc-files.md:22]: This link is meant to preserve/transfer the SEO ranking for the now separate guide on internals.

**Review by shcheklein — APPROVED** [review]

**Comment by shcheklein** [comment]
there are failing links on CI^?

**Comment by jorgeorpinel** [comment]
p.s. I almost forgot to update `content/linked-terms.js` and to add a `redirects-list.json`!

CI checks are passing now BTW 🙂

**Comment by jorgeorpinel** [comment]
Actually it's skipping the [links check](https://github.com/iterative/dvc.org/pull/2043/checks?check_run_id=1592375890) now, idk why. But I did fix some more links from the blog.

**Comment by jorgeorpinel** [comment]
OK I see the [failed links](https://github.com/iterative/dvc.org/pull/2043/checks?check_run_id=1592394096) now. These are fine, as they don't exist on prod yet. But they will soon, merging! (I did a sanity check of the redirect and `.dvc`, `dvc.yaml` auto-links in https://dvc-landing-dvc-yaml-pa-e4wbve.herokuapp.com/doc/user-guide/dvc-files-and-directories

**Comment by jorgeorpinel** [comment]
This also "applies" to v1 but no need to backport.

I guess we will need good reasons to backport things to `v1` e.g. existing PRs (pre-2.0) and bug fixes.

**Comment by jorgeorpinel** [comment]
Hmmmm... https://dvc.org/doc/user-guide/dvc-files-and-directories is giving me a 404 but the redirect is there:

https://github.com/iterative/dvc.org/blob/ecc07086f686488c4933eecce7bcaee4ba84d8ee/redirects-list.json#L32

> I tested in the review app before merging.

**Pull request merged**